sycl/test-e2e/NewOffloadDriver/aot-cpu.cpp

```diff
@@ -0,0 +1,6 @@
+// REQUIRES: opencl-aot, cpu
+
+// Test with `--offload-new-driver`
+// RUN: %clangxx -fsycl -fsycl-device-code-split=per_source -fsycl-targets=spir64_x86_64 -I %S/Inputs -o %t.out %S/split-per-source-main.cpp %S/Inputs/split-per-source-second-file.cpp \
+// RUN: -fsycl-dead-args-optimization --offload-new-driver
```

Comment:
`-fsycl-dead-args-optimization` seems to be enabled by default, so do we really need to pass it everywhere?

Reply:
I have lifted this file again from ./sycl/test-e2e/DeviceCodeSplit/aot-cpu.cpp. May be we can have a cleanup PR later on?